### ISSUE TITLE
Harmonize AllZip with All

### ISSUE THREAD

**Issue by treeowl** [well-typed/generics-sop#138]
`All` comes with a method to use its constraints recursively; `AllZip` does not. Presumably one of those designs is better and should be used for both.

**Comment by kosmikus** [comment]
Presumably, you mean adding a function like `cpara_SList` to `AllZip`?

I agree that this would be good as well.

In principle, I'd even like to just make `AllZip c xs ys` a synonym for `All (Uncurry c) (Zip xs ys)` (which would explain the name choice), but I currently don't see a way to do that while retaining the `SameShapeAs` trick which improves inference.

**Comment by treeowl** [comment]
Yes, such a function, making sure it can build the necessary evidence.

**Comment by kosmikus** [comment]
I've done this in #142. I haven't yet tested some other libs whether they compile ok with this change.
